raiden/api/rest.py

```diff
@@ -413,11 +417,14 @@ def patch_channel(self, channel_address, balance=None, state=None):
                     "Can't deposit on a closed channel",
                     httplib.CONFLICT,
                 )
-            raiden_service_result = self.raiden_api.deposit(
-                channel.token_address,
-                channel.partner_address,
-                balance
-            )
+            try:
+                raiden_service_result = self.raiden_api.deposit(
+                    channel.token_address,
+                    channel.partner_address,
+                    balance
+                )
+            except InsufficientFunds as e:
```

Comment:
Also please add either the 409 Conflict or 402 Payment needed status code in the documentation of possible deposit return status code along with an explanation of when it happens: https://github.com/raiden-network/raiden/blob/459c13e742e84441e7f3a6dab2fe46b4d9d8b6e8/docs/Rest-Api.rst#possible-responses-7